os/osx/dnd.h

```diff
@@ -0,0 +1,129 @@
+// LAF OS Library
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef OS_OSX_DND_H_INCLUDED
+#define OS_OSX_DND_H_INCLUDED
+#pragma once
+
+#ifdef __OBJC__
+
+#include "base/exception.h"
+#include "base/fs.h"
+#include "base/paths.h"
+#include "clip/clip.h"
+#include "clip/clip_osx.h"
+#include "gfx/point.h"
+#include "os/dnd.h"
+#include "os/surface.h"
+#include "os/surface_format.h"
+#include "os/system.h"
+
+#include <Cocoa/Cocoa.h>
+#include <memory>
+
+namespace os {
+  class DragDataProviderOSX : public DragDataProvider {
```

Comment:
Move the implementation of this class and functions inside this file in a new .mm file.